webbpsf/wfirst.py

```diff
@@ -384,29 +565,19 @@ def pupil_mask(self, name):
                 - "UNMASKED":
                     Unmasked pupil override
         """
-
-        if name and isinstance(name, str):
-            name = name.upper()
-            if "AUTO" == name:
-                self.auto_pupil = True
-                _log.info("Using default pupil mask.")
-            elif "COLD_PUPIL" == name:
-                self.auto_pupil = False
-                _log.info("Using custom pupil mask: Masked Pupil.")
-                self.pupil = self._masked_pupil_path
-            elif "UNMASKED" == name:
-                self.auto_pupil = False
-                _log.info("Using custom pupil mask: Unmasked Pupil.")
-                self.pupil = self._unmasked_pupil_path
-            else:
-                raise ValueError("Instrument {0} doesn't have a pupil mask called '{1}'.".format(self.name, name))
-        else:
-            raise ValueError("Pupil mask setting is not valid or empty.")
-        self._pupil_mask = name
+        self._pupil_controller.pupil_mask = name
 
     def _addAdditionalOptics(self, optsys, **kwargs):
         return optsys, False, None
 
+    @property
+    def _unmasked_pupil_path(self):
```

Comment:
What are these hidden properties needed for? Is there a way the code could be simplified to not need these defined on the WFI class in duplication of the pupil controller class?